libraries/Microsoft.Bot.Builder.Core.Extensions/RecognizerResult.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+
+namespace Microsoft.Bot.Builder.Core.Extensions
+{
+    /// <summary>
+    /// Recognizer return value.
+    /// </summary>
+    public class RecognizerResult
+    {
+        /// <summary>
+        /// Original text to recognizer.
+        /// </summary>
+        public string Text { set; get; }
+
+        /// <summary>
+        /// Text modified by recognizer for example by spell correction.
+        /// </summary>
+        public string AlteredText { set; get; }
+
+        /// <summary>
+        /// Object with the intent as key and the confidence as value.
+        /// </summary>
+        public JObject Intents { get; set; }
```

Comment:
I'm not nearly as comfortable with the Recognizer Result here. Returning JObjects from our core SDK seems like a big abstraction leak. Let's sync on this before we commit.